### ISSUE TITLE
Warning for disable directives that do nothing.

### ISSUE THREAD

**Issue by BobCga** [koalaman/shellcheck#645]
It should be possible to detect and report when a disable directive is doing nothing.  There are at least two ways to approach this:

1. When a shellcheck disable directive is present, but the disabled check isn't raised by the following line (or in the file when the directive is at the top), then emit a warning something like:

     "Line nn: 'SCnnnn' disabled but not raised. This shellcheck directive should be removed."

2. Provide a shellcheck option to ignore or validate disable directives.

**Comment by matthewpersico** [comment]
+1 - I was just about to enter a request for this, but github is apparently beta-ing a feature that pops up possibly-related issues to what you type for an issue title and this popped. Nice github feature. Please consider warning on directives that may no longer apply to the code that follows them. I've been going back and correcting code and forgetting to whack the directives.

**Comment by LLyaudet** [comment]
Hello,

Slightly more detailed specifications for the feature:

#### Here's a snippet or screenshot that shows a potential problem:

```sh
#!/bin/bash

# shellcheck disable=SC1000
a="WAT"

echo "${a}"
```

#### Here's what shellcheck currently says:
No warning at all.


#### Here's what I wanted to see:
A warning for SC9999, useless-suppression like in pylint:
```
Warning SC9999:  useless-suppression
The suppression for SC1000 at line 3 is useless:
# shellcheck disable=SC1000
No violation of SC1000 is reported if you remove it.
```
